tensorflow_addons/losses/contrastive.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Implements contrastive loss."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow_addons.utils import keras_utils
+
+
+@keras_utils.register_keras_custom_object
+@tf.function
+def contrastive_loss(y_true, y_pred, margin=1.0):
+    """Computes the contrastive loss.
+
+    This loss encourages the embedding to be close to each other for
+    the samples of the same label and the embedding to be far apart at least
+    by the margin constant for the samples of different labels.
+
+    See: http://yann.lecun.com/exdb/publis/pdf/hadsell-chopra-lecun-06.pdf
+
+    Args:
+      y_true: 1-D integer `Tensor` with shape [batch_size] of
+        binary labels indicating positive vs negative pair.
+      y_pred: 1-D float `Tensor` with shape [batch_size] of
+        distances between two embedding matrices.
```

Comment:
I'm afraid that we have to explain how to calculate the embedding distance `y_pred` in details, otherwise users might not get it correctly.